experimental/packages/opentelemetry-instrumentation-http/test/functionals/utils.test.ts

```diff
@@ -465,4 +465,65 @@ describe('Utility', () => {
       verifyValueInAttributes(attributes, undefined, 1200);
     });
   });
+
+  describe('headers to span attributes capture', () => {
+    let span: Span;
+
+    beforeEach(() => {
+      span = new Span(
+        new BasicTracerProvider().getTracer('default'),
+        ROOT_CONTEXT,
+        'test',
+        { spanId: '', traceId: '', traceFlags: TraceFlags.SAMPLED },
+        SpanKind.INTERNAL
+      );
+    });
+
+    it('should set attributes for request and response keys', () => {
```

Comment:
I'd like to see a test for the same header repeated:
```
Header-1: value1
Header-1: value2
```
The canonical way http wants us to represent this is as `Header-1: value1,value2` (in original order and capitalization, comma separated: https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2).  I'm open to other sane approaches, as long as we have a test verifying/documenting the behavior.

Reply:
The incoming headers are handled by Node's [HTTP module](https://nodejs.org/api/http.html#http_message_headers):
```
Duplicates of age, authorization, content-length, content-type, etag, expires, from, host, if-modified-since, if-unmodified-since, last-modified, location, max-forwards, proxy-authorization, referer, retry-after, server, or user-agent are discarded.
set-cookie is always an array. Duplicates are added to the array.
For duplicate cookie headers, the values are joined together with '; '.
For all other headers, the values are joined together with ', '.
```

Since header capture works on already parsed requests, we'll get the usual headers as a comma separated string, which when passed to header capturing, produces a single element array, e.g. `['value1,value2']`. This is allowed by the OTel [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers): `The attribute value MUST consist of either multiple header values as an array of strings or a single-item array containing a possibly comma-concatenated string, depending on the way the HTTP library provides access to headers.`

Reply:
Sold.